src/Microsoft.OData.Edm/Schema/EdmOperation.cs

```diff
@@ -29,15 +29,15 @@ protected EdmOperation(string namespaceName, string name, IEdmTypeReference retu
         {
             EdmUtil.CheckArgumentNull(namespaceName, "namespaceName");
 
-            this.ReturnType = returnType;
+            this.ReturnType = returnType == null ? null : new EdmOperationReturnType(this, returnType);

```

Comment:
What if the passed returnType is already an IEdmOperationReturnType? Should we check for that here and only create a new EdmOperationReturnType if casting returnType to IEdmOperationReturnType fails?

Reply:
I thought the code itself ok if the input "returnType" is already an "IEdmOperationReturnType"?
Because:

1. Supposed the "returnType" is from the operation itself. That's not a valid case. Because, we can't create an "EdmOperationReturnType" ahead of its declaring "EdmOperation", because, to create a "EdmOperationReturnType" should have the "EdmOperation" created first (see EdmOperationReturnType constructor). So, that's a conflict. 

2. Supposed the "returnType" is from other operation. So, the above code is ok. Because, we create a new "EdmOperationReturnType" and point the declaring operation to "this" and use the input "typeReference" as the type of the return type.

So, i want to keep the code unchanged. Please share your thoughts.